tower-load/src/instrument.rs

```diff
@@ -59,18 +56,23 @@ where
     }
 }
 
-impl<F, I, H> Future for InstrumentFuture<F, I, H>
+impl<F, I, H, T, E> Future for InstrumentFuture<F, I, H>
 where
-    F: Future,
-    I: Instrument<H, F::Item>,
+    F: Future<Output = Result<T, E>>,
+    I: Instrument<H, T>,
 {
-    type Item = I::Output;
-    type Error = F::Error;
+    type Output = Result<I::Output, E>;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let me = self.project();
 
-    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        let rsp = try_ready!(self.future.poll());
-        let h = self.handle.take().expect("handle");
-        Ok(self.instrument.instrument(h, rsp).into())
+        let rsp = match me.future.poll(cx) {
```

Comment:
Use `ready!`?

Reply:
I have been trying to avoid pulling in futures-util only for a small macro 🤷‍♂

Reply:
`ready!` is in `futures-core`